src/input.cpp

```diff
@@ -0,0 +1,396 @@
+    bool eTargetPad = lua_toboolean(L, 2);
+    unsigned short usDurationMicroSec = luaL_checkinteger(L, 3);
+    SteamInput()->TriggerHapticPulse(inputHandle, static_cast<ESteamControllerPad>(eTargetPad), usDurationMicroSec);
+    return 1;
+}
+
+// void TriggerRepeatedHapticPulse( InputHandle_t inputHandle, ESteamControllerPad eTargetPad, unsigned short usDurationMicroSec, unsigned short usOffMicroSec, unsigned short unRepeat, unsigned int nFlags );
+EXTERN int luasteam_triggerRepeatedHapticPulse(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    bool eTargetPad = lua_toboolean(L, 2);
+    unsigned short usDurationMicroSec = luaL_checkinteger(L, 3);
+    unsigned short usOffMicroSec = luaL_checkinteger(L, 4);
+    unsigned short unRepeat = luaL_checkinteger(L, 5);
+    SteamInput()->TriggerRepeatedHapticPulse(inputHandle, static_cast<ESteamControllerPad>(eTargetPad), usDurationMicroSec, usOffMicroSec, unRepeat, 0);
+    return 1;
+}
+
+// void TriggerVibration( InputHandle_t inputHandle, unsigned short usLeftSpeed, unsigned short usRightSpeed );
+EXTERN int luasteam_triggerVibration(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    unsigned short usLeftSpeed = luaL_checkinteger(L, 2);
+    unsigned short usRightSpeed = luaL_checkinteger(L, 3);
+    SteamInput()->TriggerVibration(inputHandle, usLeftSpeed, usRightSpeed);
+    return 1;
+}
+
+// EInputActionOrigin GetActionOriginFromXboxOrigin( InputHandle_t inputHandle, EXboxOrigin eOrigin );
+EXTERN int luasteam_getActionOriginFromXboxOrigin(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    EXboxOrigin eOrigin = static_cast<EXboxOrigin>(luaL_checkoption(L, 2, nullptr, xbox_origins));
+    lua_pushstring(L, input_action_origins[SteamInput()->GetActionOriginFromXboxOrigin(inputHandle, eOrigin)]);
+    return 1;
+}
+
+// EInputActionOrigin TranslateActionOrigin( ESteamInputType eDestinationInputType, EInputActionOrigin eSourceOrigin );
+EXTERN int luasteam_translateActionOrigin(lua_State *L) {
+    ESteamInputType eDestinationInputType = static_cast<ESteamInputType>(luaL_checkoption(L, 1, nullptr, steam_input_types));
+    EInputActionOrigin eSourceOrigin = static_cast<EInputActionOrigin>(luaL_checkoption(L, 2, nullptr, input_action_origins));
+    lua_pushstring(L, input_action_origins[SteamInput()->TranslateActionOrigin(eDestinationInputType, eSourceOrigin)]);
+    return 1;
+}
+
+// bool GetDeviceBindingRevision( InputHandle_t inputHandle, int *pMajor, int *pMinor );
+EXTERN int luasteam_getDeviceBindingRevision(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    int nMajorRevision = -1;
+    int nMinorRevision= -1;
+    bool succ = SteamInput()->GetDeviceBindingRevision(inputHandle, &nMajorRevision, &nMinorRevision);
+    
+    lua_createtable(L, 0, 3);
+    lua_pushboolean(L, succ);
+    lua_setfield(L, -2, "success");
+    lua_pushinteger(L, nMajorRevision);
```

Comment:
I think we use `lua_pushnumber` everywhere else. And I'm not sure how `pushinteger` works on older lua and luajit, which is what this lib is mostly used for. So might be best to use `pushnumber` everywhere. Doubles should be able to hold any int.